In [3]:
import pandas as pd
import scipy.stats as ss
import scipy.interpolate as si
import numpy as np
import seaborn as sbn
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import plotly.express as px
from plotly.subplots import make_subplots
from sklearn.preprocessing import LabelEncoder
import warnings
warnings.filterwarnings("ignore")
from iso3166 import Country
from datetime import datetime

Gerekli kütüphaneleri çalışma sayfasına yükledik.

    pandas     -- Veri Okuma ve Veri Çerçevesi gibi kolaylıklar sağlar.
    numpy      -- İstatistiksel bazı öngörüler ve işlemler için kullandık.
    seaborn    -- İstatistiksel grafikleri daha iyi şekilde sunar.
    matplotlib -- Grafik oluşturma için kullandık.
    plotly     -- Daha kapssamlı grafikleri içerdiği için kullandık.
    sklearn    -- Makine Öğenimi için kulladnığımız bir kütüphanedir. Kayıp veri için önemlidir.
    warnings   -- Hata yönetimi konusunda her hatayı göstermesin diye kullandık.
    iso3166    -- Bu paket içerisinde ülkeler şehirler gibi kolaylık sağlayan araçlar mevcuttur.
    datetime   -- Zaman serileri ile çalışmak için kullandık.



In [4]:
Data= pd.read_csv("Space_Corrected.csv",sep=",",parse_dates=["Datum"])

Data.drop(columns=["Unnamed: 0","Unnamed: 0.1"],inplace=True)

Data.head()

Company Name                                           Location  \
0       SpaceX         LC-39A, Kennedy Space Center, Florida, USA   
1         CASC  Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2       SpaceX                      Pad A, Boca Chica, Texas, USA   
3    Roscosmos       Site 200/39, Baikonur Cosmodrome, Kazakhstan   
4          ULA           SLC-41, Cape Canaveral AFS, Florida, USA   

                       Datum                                        Detail  \
0  2020-08-07 05:12:00+00:00  Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   
1  2020-08-06 04:01:00+00:00           Long March 2D | Gaofen-9 04 & Q-SAT   
2  2020-08-04 23:57:00+00:00            Starship Prototype | 150 Meter Hop   
3  2020-07-30 21:25:00+00:00  Proton-M/Briz-M | Ekspress-80 & Ekspress-103   
4  2020-07-30 11:50:00+00:00                    Atlas V 541 | Perseverance   

  Status Rocket  Rocket Status Mission  
0  StatusActive   50.0         Success  
1  StatusActive  29.75         Success  
2  StatusActive     NaN        Success  
3  StatusActive   65.0         Success  
4  StatusActive  145.0         Success

Öncelikle verilerin olduğu csv dosyasını pandas kulalnarak okuduk sonrasında bazı gereksiz sütunları sildik ve inplace kullanarak bunları değişken üzerine yazdık.sonrada verinin ilk 5 değerini görüntüledik.

In [5]:
Data.describe()

Company Name                                    Location  \
count          4324                                        4324   
unique           56                                         137   
top       RVSN USSR  Site 31/6, Baikonur Cosmodrome, Kazakhstan   
freq           1777                                         235   

                            Datum                               Detail  \
count                        4324                                 4324   
unique                       4319                                 4278   
top     2008-11-05 00:15:00+00:00  Cosmos-3MRB (65MRB) | BOR-5 Shuttle   
freq                            2                                    6   

        Status Rocket  Rocket Status Mission  
count            4324     964           4324  
unique              2      56              4  
top     StatusRetired  450.0         Success  
freq             3534     136           3879

Verinin özet istatistiklerine baktık. bu bize bi kaç şekilde kolaylık sağlar.

In [6]:
Data[['Launch Pad','Launch Center','Launch City','Launch Country']] = Data.Location.str.split(",",expand=True)

Location verilerinin olduğu sütunu parçaladık ve içinden işimize yarayan verileri aktif kullanmak içim farklı sütunlara böldük.

In [7]:
Data[['Launch Vehicle','Launch Payload or Mission']]=Data.Detail.str.split("|",expand=True)

Detail verilerinin odluğu sütunu parçaladık ve içinden bilgileri farklı sütunlara aktardık.

In [8]:
Data["Datum"]=pd.to_datetime(Data["Datum"],utc=True)

In [9]:
Data['Launch Year']=Data.Datum.dt.year.values
Data['Launch Month']=Data.Datum.dt.month
Data['Launch Day']=Data.Datum.dt.day
Data['Launch Clock']=Data.Datum.dt.time

In [10]:
Data.head()

Company Name                                           Location  \
0       SpaceX         LC-39A, Kennedy Space Center, Florida, USA   
1         CASC  Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2       SpaceX                      Pad A, Boca Chica, Texas, USA   
3    Roscosmos       Site 200/39, Baikonur Cosmodrome, Kazakhstan   
4          ULA           SLC-41, Cape Canaveral AFS, Florida, USA   

                      Datum                                        Detail  \
0 2020-08-07 05:12:00+00:00  Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   
1 2020-08-06 04:01:00+00:00           Long March 2D | Gaofen-9 04 & Q-SAT   
2 2020-08-04 23:57:00+00:00            Starship Prototype | 150 Meter Hop   
3 2020-07-30 21:25:00+00:00  Proton-M/Briz-M | Ekspress-80 & Ekspress-103   
4 2020-07-30 11:50:00+00:00                    Atlas V 541 | Perseverance   

  Status Rocket  Rocket Status Mission         Launch Pad  \
0  StatusActive   50.0         Success             LC-39A   
1  StatusActive  29.75         Success  Site 9401 (SLS-2)   
2  StatusActive     NaN        Success              Pad A   
3  StatusActive   65.0         Success        Site 200/39   
4  StatusActive  145.0         Success             SLC-41   

                      Launch Center  Launch City Launch Country  \
0              Kennedy Space Center      Florida            USA   
1   Jiuquan Satellite Launch Center        China           None   
2                        Boca Chica        Texas            USA   
3               Baikonur Cosmodrome   Kazakhstan           None   
4                Cape Canaveral AFS      Florida            USA   

        Launch Vehicle    Launch Payload or Mission  Launch Year  \
0    Falcon 9 Block 5     Starlink V1 L9 & BlackSky         2020   
1       Long March 2D           Gaofen-9 04 & Q-SAT         2020   
2  Starship Prototype                 150 Meter Hop         2020   
3     Proton-M/Briz-M    Ekspress-80 & Ekspress-103         2020   
4         Atlas V 541                  Perseverance         2020   

   Launch Month  Launch Day Launch Clock  
0             8           7     05:12:00  
1             8           6     04:01:00  
2             8           4     23:57:00  
3             7          30     21:25:00  
4             7          30     11:50:00

Datetime objesine çevirdiğimiz Datum sütununu Yıl ay gün ve saat olarak ayırdık.Şimdi gereksiz bazı sütunlar var biz ya bunları silicez yada farklı bir dataframe içine işimize yarayan sütunları çekip alıcaz. Biz Farklı bir dataframe içine Location objesini almadan devam ettil ve farklı bir dataframe ile yolumuza devam ediyoruz. Ayrıca herhangi bir hata olduğunda tekrardan eski Dataframe dönüp işlem yapabiliriz.

In [11]:
Data_Space=Data.drop(columns={'Location'},inplace=False)

Burada Rocket isminin önündeki boşluğ sildik ve yukarıda ise inplace=False diyerek Data içindeki Location kısmını bozmamış olduk.

In [12]:
Data_Space.rename(columns={' Rocket': 'Rocket'},inplace=True)

In [13]:
Data_Space.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 16 columns):
 #   Column                     Non-Null Count  Dtype              
---  ------                     --------------  -----              
 0   Company Name               4324 non-null   object             
 1   Datum                      4324 non-null   datetime64[ns, UTC]
 2   Detail                     4324 non-null   object             
 3   Status Rocket              4324 non-null   object             
 4   Rocket                     964 non-null    object             
 5   Status Mission             4324 non-null   object             
 6   Launch Pad                 4324 non-null   object             
 7   Launch Center              4324 non-null   object             
 8   Launch City                4301 non-null   object             
 9   Launch Country             1650 non-null   object             
 10  Launch Vehicle             4324 non-null   object             
 11  Laun

Şimdi bazı sütunlarda kayıp veriler mevcut bunlara çözümler üretmemiz gerekiyor.

In [14]:
Data_Space["Launch Country"].isna().value_counts()

True     2674
False    1650
Name: Launch Country, dtype: int64

In [15]:
Data_Space["Launch Country"]=Data_Space["Launch Country"].fillna(Data_Space["Launch City"])

City bölgesinden boş olan country sütununu doldurduk.

In [16]:
Data_Space["Launch Country"].unique()

array([' USA', ' China', ' Kazakhstan', ' Japan', ' Israel',
       ' New Zealand', None, ' Russia', ' France', ' Iran', ' India',
       ' New Mexico', ' North Korea', ' Pacific Missile Range Facility',
       ' Pacific Ocean', ' South Korea', ' Barents Sea', ' Brazil',
       ' Gran Canaria', ' Kenya', ' Australia'], dtype=object)

In [17]:
Data_Space["Launch Country"].isna().value_counts()

False    4301
True       23
Name: Launch Country, dtype: int64

burada 23 adet boş değerimiz kaldı. 4301 veri dolu kullanılabilir olarak dolduruldu.

In [18]:
countries_dict = {
    ' Russia' : 'Russian Federation',
    ' New Mexico' : 'USA',
    " Yellow Sea": 'China',
    " Pacific Missile Range Facility": 'USA',
    " Barents Sea": 'Russian Federation',
    " Gran Canaria": 'USA',
    " Kazakhstan": 'Russian Federation',
    " USA":"USA",
    ' China':"China",
    " Japan":"Japan",
    " Israel":"Israel",
    ' France':"France",
    ' Iran':"Iran",
    ' India':"Iran",
    ' North Korea':"Nort Korea",
    ' South Korea':"South Korea",
    ' Brazil':"Brazil",
    ' Kenya':"USA",
    ' Australia':"Europa Company",
    ' Pacific Ocean':"Global Company",
    ' New Zealand':"USA"
}

Data_Space['Launch Country'] = Data_Space['Launch Country'].replace(countries_dict)


In [19]:
Data_Space["Launch Country"].unique()

array(['USA', 'China', 'Russian Federation', 'Japan', 'Israel', None,
       'France', 'Iran', 'Nort Korea', 'Global Company', 'South Korea',
       'Brazil', 'Europa Company'], dtype=object)

In [20]:
Data_Space[["Launch City","Launch Country"]]

Launch City      Launch Country
0         Florida                 USA
1           China               China
2           Texas                 USA
3      Kazakhstan  Russian Federation
4         Florida                 USA
...           ...                 ...
4319      Florida                 USA
4320      Florida                 USA
4321      Florida                 USA
4322   Kazakhstan  Russian Federation
4323   Kazakhstan  Russian Federation

[4324 rows x 2 columns]

In [21]:
rocket_dict={'50.0 ':'50.0','29.75 ':'29.75','5,000.0 ':'5.0','1,160.0 ':'1.16'}

Data_Space['Rocket'] = Data_Space['Rocket'].replace(rocket_dict)

Data_Space["Rocket"]=Data_Space["Rocket"].astype("float64",errors='raise')

In [22]:
Data_Space["Rocket"].isna().value_counts()

True     3360
False     964
Name: Rocket, dtype: int64

In [23]:
Unique_Company=pd.unique(Data_Space["Company Name"])
print("Unique Company Sayısı:{}".format(len(Unique_Company)))
Unique_Company[:20]

Unique Company Sayısı:56


array(['SpaceX', 'CASC', 'Roscosmos', 'ULA', 'JAXA', 'Northrop', 'ExPace',
       'IAI', 'Rocket Lab', 'Virgin Orbit', 'VKS RF', 'MHI', 'IRGC',
       'Arianespace', 'ISA', 'Blue Origin', 'ISRO', 'Exos', 'ILS',
       'i-Space'], dtype=object)

Yukarıda Eşsiz olarak bulunan kuruluşları ve sayıları bulduk. bu bize aşağıdaki işlemler için yarar sağlayacak.

In [24]:
Data_Gecici=Data_Space.copy()
Data_Space_Rocket=["Rocket"]
for e in Unique_Company:
    Company_Filtre=Data_Gecici["Company Name"]==e
    Data_Filtre=Data_Gecici[Company_Filtre]
    for s in Data_Space_Rocket:
        Ortalama=np.round(np.mean(Data_Filtre[s]),2)
        if ~np.isnan(Ortalama):
            Data_Filtre[s]=Data_Filtre[s].fillna(Ortalama)
        else:
            Tum_Ortalama=np.round(np.mean(Data_Space[s]),2)
            Data_Filtre[s]=Data_Filtre[s].fillna(Tum_Ortalama)
    Data_Gecici[Company_Filtre]=Data_Filtre
Data_Space=Data_Gecici.copy()
Data_Space.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4324 entries, 0 to 4323
Data columns (total 16 columns):
 #   Column                     Non-Null Count  Dtype              
---  ------                     --------------  -----              
 0   Company Name               4324 non-null   object             
 1   Datum                      4324 non-null   datetime64[ns, UTC]
 2   Detail                     4324 non-null   object             
 3   Status Rocket              4324 non-null   object             
 4   Rocket                     4324 non-null   float64            
 5   Status Mission             4324 non-null   object             
 6   Launch Pad                 4324 non-null   object             
 7   Launch Center              4324 non-null   object             
 8   Launch City                4301 non-null   object             
 9   Launch Country             4301 non-null   object             
 10  Launch Vehicle             4324 non-null   object             
 11  Laun

Burada her bir Kurulus bünyesinde Nan olmayan değerlerin ortalamaları ile yine aynı kurluşun Nan değerleri ile doldurduk. Eğer Kuruluş bünyesinde bulunan veriler hepsi nan veri ise onları da tum verinin ortalaması ile doldurduk. böylece daha efektif bir şekilde kayıp verimizi düzenlemş olduk.

In [25]:
Data_Space.dropna(axis=0,how="any",inplace=True)
Data_Space.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4301 entries, 0 to 4323
Data columns (total 16 columns):
 #   Column                     Non-Null Count  Dtype              
---  ------                     --------------  -----              
 0   Company Name               4301 non-null   object             
 1   Datum                      4301 non-null   datetime64[ns, UTC]
 2   Detail                     4301 non-null   object             
 3   Status Rocket              4301 non-null   object             
 4   Rocket                     4301 non-null   float64            
 5   Status Mission             4301 non-null   object             
 6   Launch Pad                 4301 non-null   object             
 7   Launch Center              4301 non-null   object             
 8   Launch City                4301 non-null   object             
 9   Launch Country             4301 non-null   object             
 10  Launch Vehicle             4301 non-null   object             
 11  Laun

In [26]:
Status_dict={'StatusRetired': 'Status Retired',
            'StatusActive': 'Status Active',
            }

Data_Space['Status Rocket']=Data_Space['Status Rocket'].replace(Status_dict)

In [27]:
Data_Space

Company Name                     Datum  \
0          SpaceX 2020-08-07 05:12:00+00:00   
1            CASC 2020-08-06 04:01:00+00:00   
2          SpaceX 2020-08-04 23:57:00+00:00   
3       Roscosmos 2020-07-30 21:25:00+00:00   
4             ULA 2020-07-30 11:50:00+00:00   
...           ...                       ...   
4319      US Navy 1958-02-05 07:33:00+00:00   
4320         AMBA 1958-02-01 03:48:00+00:00   
4321      US Navy 1957-12-06 16:44:00+00:00   
4322    RVSN USSR 1957-11-03 02:30:00+00:00   
4323    RVSN USSR 1957-10-04 19:28:00+00:00   

                                            Detail   Status Rocket  Rocket  \
0     Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   Status Active   50.00   
1              Long March 2D | Gaofen-9 04 & Q-SAT   Status Active   29.75   
2               Starship Prototype | 150 Meter Hop   Status Active   54.99   
3     Proton-M/Briz-M | Ekspress-80 & Ekspress-103   Status Active   65.00   
4                       Atlas V 541 | Perseverance   Status Active  145.00   
...                                            ...             ...     ...   
4319                     Vanguard | Vanguard TV3BU  Status Retired  127.80   
4320                           Juno I | Explorer 1  Status Retired  127.80   
4321                       Vanguard | Vanguard TV3  Status Retired  127.80   
4322                    Sputnik 8K71PS | Sputnik-2  Status Retired    5.00   
4323                    Sputnik 8K71PS | Sputnik-1  Status Retired    5.00   

     Status Mission         Launch Pad                     Launch Center  \
0           Success             LC-39A              Kennedy Space Center   
1           Success  Site 9401 (SLS-2)   Jiuquan Satellite Launch Center   
2           Success              Pad A                        Boca Chica   
3           Success        Site 200/39               Baikonur Cosmodrome   
4           Success             SLC-41                Cape Canaveral AFS   
...             ...                ...                               ...   
4319        Failure             LC-18A                Cape Canaveral AFS   
4320        Success             LC-26A                Cape Canaveral AFS   
4321        Failure             LC-18A                Cape Canaveral AFS   
4322        Success           Site 1/5               Baikonur Cosmodrome   
4323        Success           Site 1/5               Baikonur Cosmodrome   

      Launch City      Launch Country       Launch Vehicle  \
0         Florida                 USA    Falcon 9 Block 5    
1           China               China       Long March 2D    
2           Texas                 USA  Starship Prototype    
3      Kazakhstan  Russian Federation     Proton-M/Briz-M    
4         Florida                 USA         Atlas V 541    
...           ...                 ...                  ...   
4319      Florida                 USA            Vanguard    
4320      Florida                 USA              Juno I    
4321      Florida                 USA            Vanguard    
4322   Kazakhstan  Russian Federation      Sputnik 8K71PS    
4323   Kazakhstan  Russian Federation      Sputnik 8K71PS    

        Launch Payload or Mission  Launch Year  Launch Month  Launch Day  \
0       Starlink V1 L9 & BlackSky         2020             8           7   
1             Gaofen-9 04 & Q-SAT         2020             8           6   
2                   150 Meter Hop         2020             8           4   
3      Ekspress-80 & Ekspress-103         2020             7          30   
4                    Perseverance         2020             7          30   
...                           ...          ...           ...         ...   
4319               Vanguard TV3BU         1958             2           5   
4320                   Explorer 1         1958             2           1   
4321                 Vanguard TV3         1957            12           6   
4322                    Sputnik-2         1957            11           3   
4323     

In [28]:
Data_Space.to_csv("Space_Mission_Data_Cleanning.csv",index=False)

In [29]:
Year_list = list(Data_Space['Launch Year'].unique())

num_launch = []
num_success = []
num_failure = []
prob_success = []
prob_failure = []
for n in Year_list:
    num_launch.append(((Data_Space[Data_Space['Launch Year']==n]).shape)[0])
    num_success.append((Data_Space[(Data_Space['Launch Year']==n) & (Data_Space['Status Mission'] == 'Success')]).shape[0])
    num_failure.append((Data_Space[(Data_Space['Launch Year']==n) & (Data_Space['Status Mission'] == 'Failure')]).shape[0])

for m in range(len(num_launch)):
    prob_success.append(num_success[m]/num_launch[m])
    prob_failure.append(num_failure[m]/num_launch[m])

data = {'Year': Year_list, 'Launchs': num_launch, 'Success': num_success, 'Failure':num_failure,
        'Success Probability': prob_success,'Failure Probability':prob_failure}


Space_Year= pd.DataFrame(data=data, columns= ['Year', 'Launchs', 'Success','Failure','Success Probability','Failure Probability'])


Space_Year.nlargest(20,columns=['Year'], keep='first')


Year  Launchs  Success  Failure  Success Probability  Failure Probability
0   2020       61       55        6             0.901639             0.098361
1   2019      106       97        6             0.915094             0.056604
2   2018      115      111        2             0.965217             0.017391
3   2017       91       84        5             0.923077             0.054945
4   2016       90       86        2             0.955556             0.022222
5   2015       52       48        3             0.923077             0.057692
6   2014       53       51        1             0.962264             0.018868
7   2013       45       42        2             0.933333             0.044444
8   2012       38       34        3             0.894737             0.078947
9   2011       42       40        1             0.952381             0.023810
10  2010       37       34        3             0.918919             0.081081
11  2009       50       47        3             0.940000             0.060000
12  2008       48       45        2             0.937500             0.041667
13  2007       50       46        2             0.920000             0.040000
14  2006       48       45        3             0.937500             0.062500
15  2005       37       34        3             0.918919             0.081081
16  2004       40       37        1             0.925000             0.025000
17  2003       51       48        2             0.941176             0.039216
18  2002       48       47        1             0.979167             0.020833
19  2001       42       39        1             0.928571             0.023810

In [30]:
Company_list = list(Data_Space['Company Name'].unique())

num_launch = []
num_success = []
num_failure = []
prob_success = []
prob_failure = []
for n in Company_list:
    num_launch.append(((Data_Space[Data_Space['Company Name']==n]).shape)[0])
    num_success.append((Data_Space[(Data_Space['Company Name']==n) & (Data_Space['Status Mission'] == 'Success')]).shape[0])
    num_failure.append((Data_Space[(Data_Space['Company Name']==n) & (Data_Space['Status Mission'] == 'Failure')]).shape[0])
    
for m in range(len(num_launch)):
    prob_success.append(num_success[m]/num_launch[m])
    prob_failure.append(num_failure[m]/num_launch[m])

data = {'Company': Company_list, 'Launchs': num_launch, 'Success': num_success, 'Failure':num_failure,
        'Success Probability': prob_success,'Failure Probability':prob_failure}


Space_Company= pd.DataFrame(data=data, columns= ['Company', 'Launchs', 'Success','Failure','Success Probability','Failure Probability'])


Space_Company.nlargest(40,columns=['Success Probability'], keep='first')


Company  Launchs  Success  Failure  Success Probability  \
4               JAXA        5        5        0             1.000000   
14       Blue Origin       12       12        0             1.000000   
18           i-Space        1        1        0             1.000000   
23             CASIC        2        2        0             1.000000   
27        Khrunichev        1        1        0             1.000000   
37           Starsem        1        1        0             1.000000   
43          Yuzhmash        2        2        0             1.000000   
44           Douglas        1        1        0             1.000000   
45               ASI        9        9        0             1.000000   
50           OKB-586        2        2        0             1.000000   
3                ULA      140      139        0             0.992857   
17               ILS       46       45        0             0.978261   
12       Arianespace      279      269        7             0.964158   
32            Boeing      136      131        3             0.963235   
26         Kosmotras       22       21        1             0.954545   
11               MHI       84       80        2             0.952381   
0             SpaceX      100       94        4             0.940000   
35          Lockheed       79       74        5             0.936709   
10            VKS RF      201      188        7             0.935323   
2          Roscosmos       55       51        3             0.927273   
21          Eurockot       13       12        1             0.923077   
1               CASC      245      225       14             0.918367   
28        Sea Launch       36       33        3             0.916667   
31              NASA      203      186       11             0.916256   
38         RVSN USSR     1777     1614      121             0.908272   
5           Northrop       83       74        7             0.891566   
6             ExPace        9        8        1             0.888889   
42   Martin Marietta      114      100       11             0.877193   
33              ISAS       30       26        3             0.866667   
22       Land Launch        7        6        0             0.857143   
8         Rocket Lab       13       11        2             0.846154   
15              ISRO       76       63        8             0.828947   
7                IAI       11        9        2             0.818182   
41  General Dynamics      251      203       37             0.808765   
46      US Air Force      161      129       30             0.801242   
47              CNES        8        6        2             0.750000   
52   Arm??e de l'Air        4        3        0             0.750000   
30               ESA       13        9        3             0.692308   
34               SRC        3        2        1             0.666667   
40              MITT        2        1        1             0.500000   

    Failure Probability  
4              0.000000  
14             0.000000  
18             0.000000  
23             0.000000  
27             0.000000  
37             0.000000  
43             0.000000  
44             0.000000  
45             0.000000  
50             0.000000  
3              0.000000  
17             0.000000  
12             0.025090  
32             0.022059  
26             0.045455  
11             0.023810  
0              0.040000  
35             0.063291  
10             0.034826  
2              0.054545  
21             0.076923  
1              0.057143  
28             0.083333  
31             0.054187  
38             0.068092  
5              0.084337  
6              0.111111  
42             0.096491  
33             0.100000  
22             0.000000  
8              0.153846  
15             0.105263  
7              0.181818  
41             0.147410  
46             0.186335  
47             0.250000  
52             0.000000  
30             0.230769  
34             0.333333  
40             0.500

In [31]:
Country_list = list(Data_Space['Launch Country'].unique())

num_launch = []
num_success = []
num_failure = []
prob_success = []
prob_failure = []
for n in Country_list:
    num_launch.append(((Data_Space[Data_Space['Launch Country']==n]).shape)[0])
    num_success.append((Data_Space[(Data_Space['Launch Country']==n) & (Data_Space['Status Mission'] == 'Success')]).shape[0])
    num_failure.append((Data_Space[(Data_Space['Launch Country']==n) & (Data_Space['Status Mission'] == 'Failure')]).shape[0])
    for t in Company_list:
        Data_Space[Data_Space['Company Name']]
for m in range(len(num_launch)):
    prob_success.append(num_success[m]/num_launch[m])
    prob_failure.append(num_failure[m]/num_launch[m])

data = {'Country': Country_list, 'Launchs': num_launch, 'Success': num_success, 'Failure':num_failure,
        'Success Probability': prob_success,'Failure Probability':prob_failure}


Space_Country= pd.DataFrame(data=data, columns= ['Country', 'Launchs', 'Success','Failure','Success Probability','Failure Probability'])


Space_Country.nlargest(20,columns=['Success Probability'], keep='first')


Country  Launchs  Success  Failure  Success Probability  \
5               France      303      285       13             0.940594   
3                Japan      119      111        5             0.932773   
8       Global Company       36       33        3             0.916667   
1                China      259      236       17             0.911197   
2   Russian Federation     2094     1908      135             0.911175   
0                  USA     1373     1208      131             0.879825   
4               Israel       11        9        2             0.818182   
6                 Iran       89       67       16             0.752809   
11      Europa Company        6        3        3             0.500000   
7           Nort Korea        5        2        3             0.400000   
9          South Korea        3        1        2             0.333333   
10              Brazil        3        0        2             0.000000   

    Failure Probability  
5              0.042904  
3              0.042017  
8              0.083333  
1              0.065637  
2              0.064470  
0              0.095412  
4              0.181818  
6              0.179775  
11             0.500000  
7              0.600000  
9              0.666667  
10             0.666667